src/ws/cockpitauth.c

```diff
@@ -616,7 +618,8 @@ reply_authorize_challenge (CockpitSession *session)
   if (cockpit_authorize_type (session->authorization, &authorization_type) &&
       (g_str_equal (authorize_type, "*") || g_str_equal (authorize_type, authorization_type)))
     {
-      send_authorize_reply (session->transport, cookie, session->authorization);
+      const gchar *remote_peer = cockpit_creds_get_rhost (cockpit_web_service_get_creds (session->service));
```

Comment:
(thinking out loud)

The previous code path checked this value for `NULL`.  You don't.

...but I guess that's OK because the parameter to `send_authorize_reply()` can be NULL...
...which effectively means the same thing, I think?

Except not.  Because cockpit-session will abort if that variable isn't present on the authorize message here.

So where's the discrepancy?  Why did we check this value for NULL before, and why did we accept a possible `NULL` return value from `getenv()` in `cockpit-session` (using `""` as a default)?  What are the exact situations where `cockpit_creds_get_rhost()` returns `NULL`?  Maybe some scenarios where the information was not forwarded from `cockpit-tls` or some reverse-proxy header fields were missing?

I bet it would be possible to conjure up a test case that would cause issues here...

Reply:
Thanks for spotting! Resolving in favor of your other [thread below](https://github.com/cockpit-project/cockpit/pull/16808#discussion_r1837932057).